packages/framework-provider-azure-infrastructure/src/infrastructure/synth/terraform-event-hub.ts

```diff
@@ -0,0 +1,35 @@
+import { eventhub } from '@cdktf/provider-azurerm'
+import { toTerraformName } from '../helper/utils'
+import { ApplicationSynthStack } from '../types/application-synth-stack'
+import { BoosterConfig } from '@boostercloud/framework-types'
+
+export class TerraformEventHub {
+  private static readonly DEFAULT_PARTITION_COUNT = 1
+
+  private static readonly DEFAULT_MESSAGE_RETENTION = 1
+
+  static build(
+    {
+      terraformStack,
+      azureProvider,
+      appPrefix,
+      resourceGroupName,
+      eventHubNamespace,
+      eventHubName,
+    }: ApplicationSynthStack,
+    config: BoosterConfig
+  ): eventhub.Eventhub {
+    if (!eventHubNamespace) {
+      throw new Error('Undefined eventHubNamespace resource')
+    }
+    const idApiManagement = toTerraformName(appPrefix, 'eh')
+    return new eventhub.Eventhub(terraformStack, idApiManagement, {
+      name: eventHubName,
+      resourceGroupName: resourceGroupName,
+      provider: azureProvider,
+      namespaceName: eventHubNamespace.name,
+      partitionCount: config.eventStreamConfiguration.parameters?.partitionCount ?? this.DEFAULT_PARTITION_COUNT, // Changing this will force-recreate the resource. Cannot be changed unless Eventhub Namespace SKU is Premium
```

Comment:
Do you know what would happen if the resource is recreated? I mean, imagine we have a partition count of 1 and now we want to us 3. The resource will be created but... will some events be lost? I know the events are already in the event-store, but we still need to deliver them to the event processor. Do you know if the "recreation" will happen flawlessly?

Reply:
It looks like this issue was solved [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/15088). It should be possible to increase the partition number. I will update the comment.

One thing about increasing the partition number is (from the Azure documentation):

```
If you use partition key with your producer applications and depend on key hashing to ensure ordering in a partition, dynamically adding partitions isn't recommended.

 Important

While the existing data preserves ordering, partition hashing will be broken for messages hashed after the partition count changes due to addition of partitions.
```

Reply:
@gonzalogarciajaubert Thanks! It would be grate to have this in the Booster documentation, pointing to that Azure link